src/packages/core/property-editor/uis/tree-picker/config/filter/property-editor-ui-tree-picker-filter.element.ts

```diff
@@ -0,0 +1,95 @@
+import { UMB_DATA_TYPE_WORKSPACE_CONTEXT } from '../../../../../data-type/workspace/data-type-workspace.context.js';
+import { UmbDocumentTypeInputElement } from '../../../../../../documents/document-types/components/document-type-input/document-type-input.element.js';
+import { UmbMediaTypeInputElement } from '../../../../../../media/media-types/components/media-type-input/media-type-input.element.js';
+import { StartNode } from '@umbraco-cms/backoffice/components';
+import type { UmbPropertyEditorUiElement } from '@umbraco-cms/backoffice/extension-registry';
+import { customElement, html, property } from '@umbraco-cms/backoffice/external/lit';
+import { UmbTextStyles } from '@umbraco-cms/backoffice/style';
+import { UmbLitElement } from '@umbraco-cms/internal/lit-element';
+
+@customElement('umb-property-editor-ui-tree-picker-filter')
```

Comment:
I had a really hard time getting the context/meaning of this name, so I think it would be good to give it some focus.

My usual go about this is to mention some different perspectives, to let you into my thinking:

So this one can pick a Type, can so far it can only pick content-types. We have a few other property editor that can pick something, we usually name them Picker.
So I was thinking something along this line:

`umb-property-editor-ui-treepicker-content-type-picker`.

( I put treepicker in there as this is very specific to this one case )

Let me know what you think — We might also need to rename the StartNode one os they are aligned :-)

Reply:
@nielslyngsoe I agree, I wasn't completely happy with naming this as `-filter`.

> Note, in the codebase `tree-picker` is hyphenated.

I'm wondering if the `-start-node` should become `-source`, _(I'm unsure about also adding `-picker` here too? 🤔),_ so `umb-property-editor-ui-tree-picker-source`, and then the `-filter` could become `umb-property-editor-ui-tree-picker-source-type-picker`.

Feels quite verbose, but yes "naming things is hard" 😅

Reply:
I've renamed `umb-property-editor-ui-tree-picker-start-node` to `umb-property-editor-ui-tree-picker-source-picker`;
and renamed `umb-property-editor-ui-tree-picker-filter` to `umb-property-editor-ui-tree-picker-source-type-picker`.

I've also aligned the relevant class/type names.